pkg/net/net.go

```diff
@@ -237,8 +225,7 @@ func (o *Overlay) handleJoin(s network.Stream) {
 			sort.Sort(neighbors)
 			o.n.Evict(o.ctx(), neighbors[len(neighbors)-1].PeerID)
 		} else {
-			rand.Seed(time.Now().Unix())
-			o.n.Evict(o.ctx(), neighbors[rand.Intn(len(neighbors))].PeerID)
+			o.n.Evict(o.ctx(), neighbors[o.r.Intn(len(neighbors))].PeerID)
```

Comment:
🐞  - `Intn` will panic if the neighborhood is empty.

Reply:
On the other hand,  neighbors[o.r.Intn(len(neighbors))].PeerID will still raise an error if the neighbors amount changes after the o.n.MaxLen() <= o.n.Len() but before calling `o.n.Records()`, unless we re-check neighbors amount. So, we could avoid this double checking by implementing a wrapper on the neighborhood, for providing atomic transactions.